documentation/src/docs/watermarking.md

```diff
@@ -0,0 +1,424 @@
+# Pravega Watermarking
+  * [Introduction](#introduction)
+  * [Overview](#overview)
+  * [API Changes](#api-changes)
+     - [Client Factory Changes](#client-factory-changes)  
+     - [Event Writer API Changes](#event-writer-api-changes)
+     - [Event Reader API Changes](#event-reader-api-changes)
+ * [Internal Changes](#internal-changes)
+     - [Server Changes](#server-changes)
+     - [Client changes](#client-changes)
+     - [Controller Changes](#controller-changes)
+        - [Distributing Streams Across Controller Instances](#distributing-streams-across-controller-instances)
+        - [Metadata](#metadata)
+        - [In-memory State](#in-memory-state)
+        - [Writer Marks](#writer-marks)
+        - [Periodic Watermark Workflow](#periodic-watermark-workflow)
+  * [Compatibility and Migration](#compatibility-and-migration)
+  * [Future Possibilities](#future-possibilities)
+  * [Disadvantages](#disadvantages)
+  * [Discarded Approaches](#discard-approaches)
+  * [References](#references)
+
+## Introduction
+The Goal of watermarking is to provide a time bound from the Writers to the Readers so that they can identify where they are in the Stream. The bound should always be conservative. Meaning that if the Writer’s time is well behaved, the mark should indicate to the Reader that it has received everything before a `timestamp`. (It may well have received more than this.)
+
+This immediately raises the issue of how **time** is defined and what properties does it have. This proposal avoids answering that question by leaving it up to the application. This decision was made for two reasons:
```

Comment:
`This proposal` looks like a copy-paste error from the PDP

Reply:
changed